Philips.CodeAnalysis.SecurityAnalyzers/LicenseAnalyzer.cs

```diff
@@ -1,43 +1,687 @@
 ﻿// © 2025 Koninklijke Philips N.V. See License.md in the project root for license information.
 
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Philips.CodeAnalysis.Common;
 
 namespace Philips.CodeAnalysis.SecurityAnalyzers
 {
+	/// <summary>
+	/// Represents a package from project.assets.json
+	/// </summary>
+	internal sealed class PackageInfo
+	{
+		public string Name { get; set; }
+		public string Version { get; set; }
+		public string Type { get; set; }
+		public string Path { get; set; }
+	}
+
 	[DiagnosticAnalyzer(LanguageNames.CSharp)]
-	public class LicenseAnalyzer : SingleDiagnosticAnalyzer
+	public class LicenseAnalyzer : DiagnosticAnalyzer
 	{
 		private const string Title = @"Avoid Packages with Unacceptable Licenses";
-		public const string MessageFormat = @"Package '{0}' has an unacceptable license. Consider adding to Allowed.Licenses.txt if license is acceptable.";
-		private const string Description = @"Packages with unacceptable licenses (e.g., copyleft licenses like GPL) should be reviewed before use to ensure compliance with project requirements.";
+		public const string MessageFormat = @"Package '{0}' version '{1}' has an unacceptable license '{2}'. " +
+											 @"Consider adding to Allowed.Licenses.txt if license is acceptable.";
+		private const string Description = @"Packages with unacceptable licenses (e.g., copyleft licenses like GPL) should be " +
+										   @"reviewed before use to ensure compliance with project requirements.";
 
 		public const string AllowedLicensesFileName = @"Allowed.Licenses.txt";
+		public const string LicensesCacheFileName = @"licenses.cache";
+		private const string ProjectAssetsFileName = @"project.assets.json";
+
+		// Default acceptable licenses (permissive licenses that are generally safe to use)
+		private static readonly HashSet<string> DefaultAcceptableLicenses = new(StringComparer.OrdinalIgnoreCase)
+		{
+			"MIT",
+			"Apache-2.0",
+			"BSD-2-Clause",
+			"BSD-3-Clause",
+			"ISC",
+			"Unlicense",
+			"0BSD",
+			"https://github.com/dotnet/corefx/blob/master/LICENSE.TXT",
+			"https://github.com/dotnet/standard/blob/master/LICENSE.TXT",
+			"http://go.microsoft.com/fwlink/?LinkId=329770",
```

Comment:
Sonar is stating "Using http protocol is insecure." To make this warning go away, should we remove the "http" prefix and rely on Contains logic? This would also future proof against http vs https